### PR TITLE
Deadlock localized and fixed, ++ more.

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: Blocks/Persistence/test.cc [hunk 1]
```diff
@@ -134,6 +134,69 @@ TEST(PersistenceLayer, MemoryOnly) {
   }
 }
 
+// NOTE: This test is obsolete. -- D.K.
```
Comment (mzhurovich) [inline Blocks/Persistence/test.cc]: So, remove these two tests?

File: Sherlock/test.cc [hunk 2]
```diff
@@ -394,6 +392,7 @@ TEST(Sherlock, PersistsToFile) {
   current::time::SetNow(std::chrono::microseconds(300u));
   persisted.Publish(3);
 
+  // This spin lock is unnecessary as publishing are synchronous as of now. -- D.K.
```
Comment (mzhurovich) [inline Sherlock/test.cc]: publishing **is** synchronous?

File: Sherlock/test.cc [hunk 3]
```diff
@@ -283,6 +276,7 @@ struct RecordsCollector final {
 
 }  // namespace sherlock_unittest
 
+/*
```
Comment (mzhurovich) [inline Sherlock/test.cc]: Uncomment this test?

**Comment by mzhurovich** [comment]
@dkorolev merging this and #374 - minor fixes could be added later (на скорость не влияет).

File: Blocks/Persistence/test.cc (hunk 1 above)
Comment (dkorolev) [inline Blocks/Persistence/test.cc]: :+1: https://github.com/c5t/Current/commit/80387e008a6585b0ae0374075b374c1c3017ee1f

File: Sherlock/test.cc (hunk 2 above)
Comment (dkorolev) [inline Sherlock/test.cc]: :+1: https://github.com/c5t/Current/commit/80387e008a6585b0ae0374075b374c1c3017ee1f

File: Sherlock/test.cc (hunk 3 above)
Comment (dkorolev) [inline Sherlock/test.cc]: :+1: https://github.com/c5t/Current/commit/80387e008a6585b0ae0374075b374c1c3017ee1f

**Pull request merged**